puppet/modules/deploy/files/script.sh

```diff
@@ -2,7 +2,9 @@
 set -e # dont rsync if clone fails
 echo "Deploy started at `date`"
 dir=`mktemp -d`
-trap "rm -rf $dir" EXIT
-git clone --recurse-submodules https://github.com/theforeman/foreman-infra $dir/
-rsync -aqx --delete-after --exclude=.git $dir/puppet/modules/ /etc/puppetlabs/code/environments/production/modules/
+trap "rm -rf ${dir}" EXIT
+git clone --recurse-submodules https://github.com/theforeman/foreman-infra ${dir}/
+prod_dir="/etc/puppetlabs/code/environments/production"
+rsync -aqx --delete-after --exclude=.git ${dir}/puppet/*modules ${prod_dir}/
```

Comment:
This might deleta a bit too much, but I don't know if recent puppetserver releases are tripping over non-existing `data` and `manifests` folders...

TBH, after all this wrangling I see the idea of just syncing all modules into one dir at deployment much more feasible now. ;)

Reply:
IMHO this is mostly a temporary solution and we should use r10k to deploy after this since it has caching built in as well.